cmd/snap-confine/seccomp-support.c

```diff
@@ -43,32 +45,40 @@ typedef struct sock_filter bpf_instr;
 static void validate_path_has_strict_perms(const char *path)
 {
 	struct stat stat_buf;
-	if (stat(path, &stat_buf) < 0)
+	if (stat(path, &stat_buf) < 0) {
 		die("cannot stat %s", path);
+	}
 
 	errno = 0;
-	if (stat_buf.st_uid != 0 || stat_buf.st_gid != 0)
+	if (stat_buf.st_uid != 0 || stat_buf.st_gid != 0) {
 		die("%s not root-owned %i:%i", path, stat_buf.st_uid,
 		    stat_buf.st_gid);
+	}
 
-	if (stat_buf.st_mode & S_IWOTH)
+	if (stat_buf.st_mode & S_IWOTH) {
 		die("%s has 'other' write %o", path, stat_buf.st_mode);
+	}
 }
 
 static void validate_bpfpath_is_safe(const char *path)
 {
-	if (path == NULL || strlen(path) == 0 || path[0] != '/')
+	if (path == NULL || strlen(path) == 0 || path[0] != '/') {
 		die("valid_bpfpath_is_safe needs an absolute path as input");
-
+	}
 	// strtok_r() modifies its first argument, so work on a copy
-	char *tokenized = strdup(path);
-
+	char *tokenized __attribute__ ((cleanup(sc_cleanup_string))) = NULL;
+	tokenized = strdup(path);
+	if (tokenized == NULL) {
+		die("cannot allocate memory for copy of path");
+	}
 	// allocate a string large enough to hold path, and initialize it to
 	// '/'
-	size_t checked_path_size = sizeof(char) * strlen(path) + 1;
-	char *checked_path = malloc(checked_path_size);
-	if (checked_path == NULL)
-		die("Out of memory creating checked_path");
+	size_t checked_path_size = strlen(path) + 1;
+	char *checked_path __attribute__ ((cleanup(sc_cleanup_string))) = NULL;
+	checked_path = calloc(checked_path_size, 1);
```

Comment:
I know that '1' is equivalent to `sizeof(char)`, but I think that `sizeof(char)` declares intent of use better for the calloc and to a lesser extent checked_path_size.